app/src/main/java/com/beemdevelopment/aegis/ui/views/AuditLogAdapter.java

```diff
@@ -0,0 +1,70 @@
+package com.beemdevelopment.aegis.ui.views;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.beemdevelopment.aegis.R;
+import com.beemdevelopment.aegis.database.AuditLogEntry;
+import com.beemdevelopment.aegis.vault.Vault;
+import com.beemdevelopment.aegis.vault.VaultEntry;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class AuditLogAdapter extends RecyclerView.Adapter<AuditLogHolder> {
+    private List<AuditLogEntry> _auditLogEntries;
+    private List<VaultEntry> _referencedEntries;
+
+    public AuditLogAdapter() {
+        _auditLogEntries = new ArrayList<>();
+        _referencedEntries = new ArrayList<>();
+    }
+
+    public void addAuditLog(AuditLogEntry auditLogEntry) {
+        _auditLogEntries.add(auditLogEntry);
+
+        int position = getItemCount() - 1;
+        if (position == 0) {
+            notifyDataSetChanged();
+        } else {
+            notifyItemInserted(position);
+        }
+    }
+
+    public void addReferencedEntry(VaultEntry vaultEntry) {
```

Comment:
Do we need to separately add the referenced vault entries here? Perhaps we can introduce a ``AuditLogEntryModel`` type that holds the ``AuditLogEntry`` and an optional referenced ``VaultEntry``.